### PR TITLE
Minv CelebA example update

### LINKED ISSUE #303: CelebA GAN training not working as expected
# Issue

The GAN training in the CelebA example is not working as expected, as found by Fazelehs testing

## Problem Description

We've found the following problems that needs to be adressed: 

- In train_gan() in celebA_plgmi_handler.py, the discriminator is shown augmented fake images - they are not augmented in the original paper.
- In plgmi.py, the current version of where train_gan() is called, the incorrect batch size is passed. 
- In celebA_data.py, the labels are assigned in "alphabetical" order instead of numerical order making it difficult to relate to the original CelebA training data set.  

## Expected Behavior

The example should match the GAN training method of the original PLG-MI attack

## What Needs to be Done

- Change train_gan() method in handler accordingly
- Change train_gan() call in plgmi.py accordingly
- Change celebA_data.py to la

## How Can It Be Tested or Reproduced

Run the attack and verify that the GAN training is working.

### LINKED ISSUE #304: Minv CelebA example target model metadata outdated
# Issue
## Problem Description

The resnet18 target model used in the minv CelebA example uses outdated MIA metadata, this causes schema checks to fail

## Expected Behavior

The target model metadata should follow the format in new MIA example

## What Needs to be Done

Add a separate target model class that has an updated metadata

## How Can It Be Tested or Reproduced

Train a new target model and run leakpro

### LINKED ISSUE #318: Minv Evaluation model support
# Feature

## Desired Behavior / Functionality

For evaluation of minv attacks, a separate evaluation model with the same task as the target model should be used - as  in the paper

## What Needs to Be Done

Add evaluation model support 

## How Can It Be Tested

Run example to get new results

### PR DESCRIPTION
**Pull request by oscarwernqvist**
# Description

Summary of changes

* Fixed error in GAN training loop in handler where augmentations were incorrectly used
* Fixed batch size issue in plgmi.py
* Changed target model to work with new schemas - and upgraded to ResNet152 (used in paper)
* Fixed labeling order of data in CelebA example
* Added support for PR 309 - audit as list
* Added evaluation model support - separate model for evaluation (can still be same as target model)
* Changed attack parameters to align with PLG-MI paper

## Resolved Issues

- [ ] fixes #303 
- [ ] fixes #304
- [ ] fixes #318 

## How Has This Been Tested?

- Running example works - GAN is training.
- Verified with original data set that labels match.
- ResNet152 produces better results compared to old model

### CODE REVIEW

File: examples/minv/celebA/utils/resnet_model.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+class ResNet152(BaseCNN):
+    def __init__(self, num_classes):
+        super().__init__(num_classes)
+        self.model = resnet152(weights=ResNet152_Weights.IMAGENET1K_V2)
+        # Replace the final fully-connected layer to match the desired number of classes
+        self.model.fc = nn.Linear(self.model.fc.in_features, num_classes)
+
+
+class VGG16(BaseCNN):
```
Comment (fazelehh) [inline examples/minv/celebA/utils/resnet_model.py:52]: misplaced with regard to the name of the file

**Review by fazelehh — CHANGES_REQUESTED** [review]
Nice work, just some minor comments. There is one error in the last cell of the notebook file that must be fixed before merge.

File: leakpro/input_handler/modality_extensions/image_metrics.py [hunk 2]
```diff
@@ -261,3 +324,42 @@ def get_generated_features(self, model: torch.nn.Module, transform: transforms.C
 
     pass
 
+
+    def save_images(self) -> None:
```
[the remark this answers is not in the document]
Comment (oscarwernqvist) [reply leakpro/input_handler/modality_extensions/image_metrics.py:328]: I'm not sure exactly what function you're thinking about? This is written by us

File: examples/minv/celebA/utils/resnet_model.py (hunk 1 above)
Comment (oscarwernqvist) [reply examples/minv/celebA/utils/resnet_model.py:52]: True! I haven't been able to produce good results with this implementation of VGG16, and it's not used, so I'll remove it from this PR

File: examples/minv/celebA/.gitignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
+data/*
+!data/.gitkeep
```
[the remark this answers is not in the document]
Comment (oscarwernqvist) [reply examples/minv/celebA/.gitignore:2]: Added

File: examples/minv/celebA_attributes/.gitignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
+data/*
```
[the remark this answers is not in the document]
Comment (oscarwernqvist) [reply examples/minv/celebA_attributes/.gitignore]: No, issue with my git tracking, removed now!

File: examples/minv/mimic/.gitignore [hunk 5]
```diff
@@ -0,0 +1,10 @@
+data/*
```
[the remark this answers is not in the document]
Comment (oscarwernqvist) [reply examples/minv/mimic/.gitignore]: No, issue with my git tracking, removed now!

File: examples/minv/celebA/run_example.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import os
```
[the remark this answers is not in the document]
Comment (fazelehh) [reply examples/minv/celebA/run_example.py:1]: yeah, untrack it before mergeing

File: examples/minv/celebA/main_celebA_minv.ipynb [hunk 7]
```diff
@@ -67,8 +67,8 @@
     "df.columns = ['img', 'label']\n",
```
Comment (fazelehh) [inline examples/minv/celebA/main_celebA_minv.ipynb:67]: <pre class="ql-syntax" spellcheck="false">Line #15.    dim_z = config['audit']['attack_list'][0]['dim_z'] # latent dimension
</pre><p>This will break if plgmi is not the first attack. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aidotse/LeakPro/pull/305/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Review by fazelehh — APPROVED** [review]
Last cell of the notebook file will break if there plgmi is not the first attack in the list. 
You can either fix it or plz create an issue for it, and leave it as it is now.
